ambry-api/src/main/java/com/github/ambry/store/Write.java

```diff
@@ -39,4 +41,27 @@ public interface Write {
    * @throws StoreException if store error occurs when writing into underlying interface
    */
   void appendFrom(ReadableByteChannel channel, long size) throws StoreException;
+
+  /**
+   * Appends the buffer into the underlying write interface (eg: file) asynchronously. Returns the number of bytes
+   * successfully written
+   * @param buffer The buffer from which data needs to be written from
+   * @return a {@link CompletableFuture} that will eventually contain the number of bytes written to the write interface
+   *         or the {@link StoreException} if an error occurred
+   */
+  default CompletableFuture<Integer> appendAsyncFrom(ByteBuffer buffer) {
```

Comment:
Same comment as for `MessageWriteSet` above. If implementations don't need both sync and async, then maybe subclass this interface.